### ISSUE TITLE
[otbn] What requirements do we have on the relationship between edn_rst_ni and rst_ni?

### ISSUE THREAD

**Issue by rswarbrick** [lowRISC/opentitan#8758]
I just managed to generate a test where we applied the EDN reset slightly before the main reset while we were getting data back from the EDN interface for RND.

When this happens, we see a failing assertion (SyncReqAckDataHoldDst2Src). This assertion looks like:
```systemverilog
    // DST domain shall not change data while waiting for SRC domain to latch it (together with
    // receiving ACK). It takes 2 SRC cycles for ACK to cross over from DST to SRC, and 1 SRC cycle
    // for the next REQ to cross over from SRC to DST. Assert that the data is stable during that
    // window, i.e. [-2,+1] SRC cycles around the SRC handshake.
    `ASSERT(SyncReqAckDataHoldDst2Src,
        src_req_i && src_ack_o |-> $past(data_o,2) == data_o && $stable(data_o) [*2],
        clk_src_i, !rst_src_ni)
```
Here, `clk_src_i` and `rst_src_ni` are the main clock and reset for OTBN (`clk_i` and `rst_ni`). The EDN clock and reset are `clk_dst_i` and `rst_dst_ni`.

The problem is that as the EDN domain goes into reset, the push/pull driver that we use in DV sets the data line to `'x`. This appears as `data_o` in the assertion above (mirrored directly from a `data_i` input). In pictures:
![image](https://user-images.githubusercontent.com/104845/138085491-d92da207-0b03-4292-8886-017222a3066e.png)

Ignoring the assertion for a second, I think the worst consequence of this for OTBN is something like "if you reset the EDN interface while there's a request being serviced, you might corrupt the random data going through the CDC". (And, of course, OTBN won't get any more random data until the EDN comes out of reset).

Is this analysis correct? (@GregAC, @imphil)

If so, do we care? As a short-term work-around, I propose to use `$assertoff` in the UVM testbench to disable this particular assertion, together with a TODO message pointing at this issue. The "proper" fix probably depends on a design decision. If we don't want to violate this assertion, I think the DV base classes might need some tweaks to make sure they don't go into reset at the wrong time.

Also @vogelpi, @msfschaffner, who have worked on the `prim_edn_req` module.

**Comment by johngt** [comment]
Discussed this internally and we believe that although the hack still exists that its behaviour is well known so marking this as a post M2 item.

**Comment by tjaychen** [comment]
this is related to #14287, which we will discuss in the silicon working group soon.

**Comment by johngt** [comment]
Check assertions to see if released

**Comment by vogelpi** [comment]
I've just discussed this with @andreaskurth . Along the discussion we had in the silicon working group and @jeoongp 's feedback in #14287 we concluded to change the assertions inside the prim_reqack_sync primitive as follows:
- If one interface is reset, also the other one needs to be reset.
- If one interface is reset, other assertions no longer need to be checked.
- After releasing the first reset, also the second reset needs to be released before the first request signal can be handled.

I plan to look into this.
